plugins/fzuse.rb

```diff
@@ -0,0 +1,150 @@
+require 'socket'
+
+# this is the main routine that's executed in the grandchild process (msfconsole -> fzf -> this)
+if $PROGRAM_NAME == __FILE__
+  exit 64 unless ARGV.length == 2
+
+  UNIXSocket.open(ARGV[0]) do |sock|
+    sock.write ARGV[1] + "\n"
+    sock.flush
+
+    puts sock.read
+  end
+  exit 0
+end
+
+module Msf
+  class Plugin::FuzzyUse < Msf::Plugin
+    class ConsoleCommandDispatcher
+      include Msf::Ui::Console::CommandDispatcher
+
+      FZF_THEME = {
+        'fg' => '-1',
+        'fg+' => 'white:regular:bold',
+        'bg' => '-1',
+        'bg+' => '-1',
+        'hl' => '-1',
+        'hl+' => 'red:regular:bold',
+        'info' => '-1',
+        'marker' => '-1',
+        'prompt' => '-1',
+        'spinner' => '-1',
+        'pointer' => 'blue:bold',
+        'header' => '-1',
+        'border' => '-1',
+        'label' => '-1',
+        'query' => '-1'
+      }.freeze
+
+      def initialize(driver)
+        super
+
+        @module_dispatcher = Msf::Ui::Console::CommandDispatcher::Modules.new(driver)
+      end
+
+      def name
+        'FuzzyUse'
+      end
+
+      #
+      # Returns the hash of commands supported by this dispatcher.
+      #
+      def commands
+        {
+          'fzuse' => 'A fuzzy use command added by the FuzzyUse plugin'
+        }
+      end
+
+      def start_pipe_server(socket_path)
+        def pipe_server(socket_path)
```

Comment:
Might be worth running rubocop on this file, as it looks like there's a nested method here